viv_utils/emulator_drivers.py

```diff
@@ -731,6 +757,26 @@ def run(self, va: int):
 
                     # stop emulating, and go to next block in the queue.
                     break
+                except envi.exc.BreakpointHit:
+                    # emulation likely wandered off, e.g., into alignment (CC bytes)
+
+                    # stop emulating, and go to next block in the queue.
+                    break
+                except BreakpointHit as e:
+                    # client indicated not to emulate beyond the given address
+                    # by raising a breakpoint (via a monitor).
+                    logger.debug(
+                        "driver: run_function: breakpoint: %s: 0x%x",
+                        e.reason,
+                        lastpc,
+                    )
+                    blockend = lastpc
+                    for mon in self._monitors:
+                        mon.postblock(self, blockstart, blockend)
+
+                    # stop emulation and done
+                    # limitation: only finds one path that leads to target address
+                    return
```

Comment:
In my use-case this works better:
1. find the first path that leads to target address `tova`
2. continue emulation from there using the DebugEmulator

Should we add a new emulator instead that does only 1.?

Reply:
yes, lets add another driver with specific behavior for tight loops.

Reply:
as a suggestion, let's split up the emulators into individual files; `emulator_drivers.py` is already over 800 lines long

Reply:
good idea

Reply:
for now all still in one file, but changes here: a55f68e
does that work?